src/vm/fieldmarshaler.cpp

```diff
@@ -3300,7 +3300,7 @@ VOID FieldMarshaler_StringUtf8::UpdateCLRImpl(const VOID *pNativeValue, OBJECTRE
         MethodDescCallSite convertToManaged(METHOD__CUTF8MARSHALER__CONVERT_TO_MANAGED);
         ARG_SLOT args[] =
         {
-            PtrToArgSlot(pNativeValue),
+            PtrToArgSlot(sz),
```

Comment:
Oh boy, silly mistake. Can you do me a favor - flip the conditional and remove the `else` since `pString` is initialized to `NULL`? Thanks.